### DISCUSSION TITLE
First Community Call

### DISCUSSION THREAD

**Discussion by bmann** [wnfs-wg/spec#22] [Community Call]
An initial kick off Community call on Wed, Aug 3rd, 9am PDT / 12pm EDT / 1800 CEST.

Register for the monthly series on Luma https://lu.ma/wnfs

## Review of all Information channels 

* GitHub org & these discussions 
* Luma for community calls and other events / calls as needed 
* WNFS channel in IPFS Discord for chat

## Introductions

Intro yourself, your project, your org, and what you need help with, are interested in working on, or anything else to share. 

Feel free to do a written intro / links in the comment thread here as well or instead. 

## Capyloon Use Case / Requirements

@fabricedesre has some specific use cases for WNFS in the context of the Capyloon mobile operating system. Review and give feedback as a group https://github.com/wnfs-wg/spec/discussions/18

## Other Working Group Kick offs?

Any other live meetings / working groups / other kick offs we need to propose / schedule?

## Add your agenda item in the comments

- more stuff from wnfs specced out with IPLD Schemas and ADLs

_we’ll edit in any other agenda items — please add in the comments_

# Recording

<a href="https://vimeo.com/manage/videos/736315713"><img width="924" alt="Screen Shot 2022-08-03 at 15 24 42" src="https://user-images.githubusercontent.com/1052016/182722841-25181a3e-743c-4373-9f23-a8ec2a1089c2.png"></a>

## Chat Log

```
00:08:23	Brooklyn Zelenka (@expede):	https://github.com/wnfs-wg/spec/discussions/22
00:08:24	Philipp Krüger:	https://github.com/wnfs-wg/spec/discussions/22
00:12:12	James Walker (@walkah):	👋
00:20:01	Boris Mann:	https://github.com/wnfs-wg/spec/discussions/18
00:28:09	Brooklyn Zelenka (@expede):	An “autocodec” would be nice for that 😉
00:32:56	Philipp Krüger:	I think the read part of tagging would work in private WNFS
00:33:05	Philipp Krüger:	But the write part would be much harder :D
00:33:24	Philipp Krüger:	I'm probably missing something of course :D
00:49:37	Philipp Krüger:	very read-side
00:52:24	Mauve:	https://github.com/ipld/go-ipld-prime/blob/master/linking/types.go#L158
00:53:06	Mauve:	https://github.com/ipld/go-ipld-prime/pull/451
01:07:04	Mauve:	Down to talk async too
```

**Comment by RangerMauve** [comment]
I'd love to talk about how we could get more stuff from wnfs specced out with IPLD Schemas and ADLs.

@aschmahmann mentioned that it could be cool to define what a filesystem even is in IPLD so that we could talk about different filesystem-like structures being interoperable.

Also, I think it'd be cool to talk about the potential for [fat pointers](https://github.com/aschmahmann/wasm-ipld/issues/6) to affect the spec (particularly when talking about symlinks)

**Reply by matheus23** [reply]
I'd love to talk to people about what their IPLD schema workflow is. I've tried to get it to work for me, but I think I got lost somewhere between trying to compile the toolchain myself and the VSCode plugin.
There [used to be `.ipldsch`](https://github.com/wnfs-wg/spec/blob/96e7f60c3c3ce61d1e3ada2eb29a6c74672e3c5b/wnfs-public.ipldsch) files in the repo (now removed and that version is outdated).
Another problem I've hit was that IPLD schemas don't really work for specifying the stuff I want to specify regarding encryption.
I'm happy to give it another try :)

**Reply by matheus23** [reply]
Regarding ADLs, I need a refresher for those. I don't quite understand how to specify them from reading the docs on https://ipld.io/docs.

Maybe someone would be interested in contributing schemas matching the public filesystem layout? (ideally after #27 is merged)

**Comment by nichoth** [comment]
> Intro yourself, your project, your org, and what you need help with, are interested in working on, or anything else to share.

[This project](https://github.com/nichoth/ssc-server#what-is-this) is designed to share images via a social network. It started with using services that are more traditional — a DB and blob host, + UCANs for ID.

In the interest of dogfood though, I've realized that what I'm really looking for is a *small scale* way to share images with specific people, and I've discovered that wnfs already has a way to do read permissions on a per-user basis. That's kind of interesting because it's all based on decryption capabilities instead of access control.

And wnfs gives us a common backend/identity source for users, which makes identity easier.

> to share private files with a group of people, you would need to write down the list of people in the group. Then call `fs.sharePrivate` and pass it the list of other users.

So wnfs I think has everything needed for this usecase. At this point it's more a matter of finding time to try this & build some UI for it. I'm coming to the call more to listen than query.

------------------------------------

relevant links:

https://guide.fission.codes/developers/webnative/sharing-private-data
https://discord.com/channels/478735028319158273/678353918752718848/996476638697099294

**Comment by RangerMauve** [comment]
What are ADLs answer:

ADLs are three things:

- A spec for what the IPLD data looks (either with JSON schemas or something else)
- A [NodeReifier](https://github.com/ipld/go-ipld-prime/blob/master/linking/types.go#L158) function which takes and IPLD node, and returns an IPLD node acts as a "lense" over the raw data
- A [Substrate](https://github.com/ipld/go-ipld-prime/blob/master/adl/interface.go#L24) property on the returned node so that applications can get the raw node back out

**Reply by RangerMauve** [reply]
https://github.com/ipld/go-ipld-prime/pull/451

**Comment by RangerMauve** [comment]
Thoughts on URLs/Protocol Handlers/ETC

It'd be useful to have a high level interface for WNFS for developers that's easy to use and easy to integrate with environments like Capyloon.

From my work with integrating p2p stuff with the web, URLs + HTTP-like interfaces (via gateways) have been really useful.

An example is that with the `ipns://` protocol handler in [Agregore](https://agregore.mauve.moe/), a developer can do a `PUT ipns://PublicKeyHere/file.txt "Hello World"` HTTP call (via window.fetch or otherwise) to upload a file and have the root IPNS entry get updated.

It'd be cool to think about a `wnfs://` protocol spec or to think about how we could use an IPLD filesystem spec to have the existing `ipfs://` handlers support wnfs.

**Comment by BigLep** [comment]
@bmann : Awesome to see this happening.  For giving community visibility to it, could meeting invites be sent to ipfs.io_eal36ugu5e75s207gfjcu0ae84@group.calendar.google.com ?  This will cause it to appear on the IPFS Community Calendar.

Context: https://github.com/ipfs/ipfs-docs/issues/1237

**Reply by BigLep** [reply]
There is discussion about a different want to give visibility to the IPFS community rather than a google calendar happening here: https://discuss.ipfs.tech/t/community-calendar-options-other-than-google-cal/14954 .  In the meantime, I have added these events to the existing IPFS Community google calendar by personally joining the Luma events and inviting the IPFS community calendar.  No further action needed at the moment.

**Comment by bmann** [comment]
We use Luma, and I don't know if an email can be added --> give it a try to see if you can self register https://lu.ma/wnfs

Also: @BigLep a longer discussion I think https://discuss.ipfs.tech/t/community-calendar-options-other-than-google-cal/14954
